### PR TITLE
[CELEBORN-468] Timeout useless lostWorkers/shutdownWorkers meta

### PR DESCRIPTION
**Pull request by RexXiong**
### What changes were proposed in this pull request?
As title

### Why are the changes needed?
If Worker lost or lost after graceful shutdown, Master would retain these lostWorker/shutdownWorkers meta permanently,
These meta would cause some noisy message in lifecycleManager. For these meta better to delete them after a while

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT & E2E test

### CODE REVIEW

File: common/src/main/scala/org/apache/celeborn/common/protocol/message/ControlMessages.scala [hunk 1]
```diff
@@ -360,6 +362,10 @@ object ControlMessages extends Logging {
         .build()
   }
 
+  case class RemoveWorkersUnavailableInfo(
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/protocol/message/ControlMessages.scala:365]: Currently it's preferred not to define case classes for control messages, use pbs instead.

File: master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java [hunk 2]
```diff
@@ -135,6 +135,17 @@ public void updateWorkerRemoveMeta(
     excludedWorkers.remove(worker);
   }
 
+  public void removeWorkersUnavailableInfoMeta(List<WorkerInfo> unavailableWorkers) {
+    synchronized (workers) {
+      for (WorkerInfo workerInfo : unavailableWorkers) {
+        if (lostWorkers.containsKey(workerInfo)) {
+          lostWorkers.remove(workerInfo);
+          shutdownWorkers.remove(workerInfo);
```
Comment (waitinfuture) [inline master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java:143]: `shutdownWorkers` do not necessarily overlap with `lostWorkers`. Need to check `shutdownWorkers` independently, or don't check contains at all.

Comment (RexXiong) [reply master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java:143]: > `shutdownWorkers` do not necessarily overlap with `lostWorkers`. Need to check `shutdownWorkers` independently, or don't check contains at all.

Only those lost workers which also in shutdownWorkers need remove.

Comment (waitinfuture) [reply master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java:143]: Right, thanks :)

**Review by waitinfuture — APPROVED** [review]
LGTM, merging to main/0.3

**Pull request closed without merging**